cfripper/rules/CrossAccountTrustRule.py

```diff
@@ -38,10 +38,21 @@ def invoke(self, cfmodel):
                         type(self).__name__, self.REASON.format(logical_id, principal), resource_ids={logical_id}
                     )
 
-                for principal in resource.Properties.AssumeRolePolicyDocument.allowed_principals_with(
-                    not_has_account_id
-                ):
-                    if not principal.endswith(".amazonaws.com"):  # Checks if principal is an AWS service
-                        self.add_failure(
-                            type(self).__name__, self.REASON.format(logical_id, principal), resource_ids={logical_id}
-                        )
+                if self._config.aws_account_id:
+                    for principal in resource.Properties.AssumeRolePolicyDocument.allowed_principals_with(
+                        not_has_account_id
+                    ):
+                        if not principal.endswith(".amazonaws.com"):  # Checks if principal is an AWS service
+                            if "GETATT" in principal or "UNDEFINED_" in principal:
```

Comment:
this should be startswith

Reply:
Yes and no. It could be possible to have: `test_UNDEFINED_PARAM_SOMETHING_role`